src/StoragelessSession/Http/SessionMiddleware.php

```diff
@@ -0,0 +1,144 @@
+<?php
+namespace StoragelessSession\Http;
+
+use DateTime;
+use Zend\Stratigility\MiddlewareInterface;
+use Psr\Http\Message\ServerRequestInterface as Request;
+use Psr\Http\Message\ResponseInterface as Response;
+use Lcobucci\JWT\Token;
+use Lcobucci\JWT\Signer;
+use Lcobucci\JWT\Builder;
+use StoragelessSession\Session\Data;
+use Lcobucci\JWT\Parser;
+use Lcobucci\JWT\ValidationData;
+use Lcobucci\JWT\Signer\Key;
+
+final class SessionMiddleware implements MiddlewareInterface
```

Comment:
Nope. We don't need an interface contract there because the middleware is not supposed to be consumed by userland code, but by a middleware dispatcher. Middleware dispatchers does not care about interfaces, they are based on **conventions**:

- https://github.com/relayphp/relayphp.github.io/blob/master/index.md#middleware-signature
- https://github.com/zendframework/zend-stratigility/blob/master/doc/book/middleware.md
- https://github.com/slimphp/Slim-Website/blob/gh-pages/docs/concepts/middleware.md#what-is-middleware